### ISSUE TITLE
`test` command's fixture is not getting checked in tests

### ISSUE THREAD

**Issue by JordanMartinez** [spacchetti/spago#138]
As discovered in #137 
- [suspected](https://github.com/spacchetti/spago/pull/137#issuecomment-472504432)
- [confirmation](https://github.com/spacchetti/spago/pull/137#issuecomment-472512322)

**Comment by JordanMartinez** [comment]
I ran the `python3 ../spago-test.py` on my local machine. I noticed that there was no `text-output.txt` file in the `spago-test` directory after a test is run. Contrast that with these files appearing:
- `spago-install-failure.dhall`
- `spago-success-failure.dhall`
- etc.

Should the `test-output.txt` file be copied over as well?

**Comment by f-f** [comment]
@JordanMartinez no, the fixture in this case is checked against the process output, code here:

https://github.com/spacchetti/spago/blob/4471eb5d3002a95bf53f5021108c17e1fa9cf326/test/utils.py#L28-L42

**Comment by JordanMartinez** [comment]
I don't know python, so I used your model to add some debug prints:
```python
# added to `call()` function
        print("Command: ")
        print(command)
        print("Failure message: ")
        print(failure_msg)
        print("Expected output fixture: ")
        print(expected_output_fixture)
        return subprocess.check_output(command, stderr=subprocess.STDOUT).decode('utf-8')
# rest of python code
```
This was the output for `test`:
```
Expecting success from: "spago test"
Command: 
['spago', 'test']
Failure message: 
Spago should test successfullytest-output.txt
Expected output fixture: 
None
```
It looks like you [forgot to add a comma at the end of this line to separate the function's arguments](https://github.com/spacchetti/spago/blob/master/test/spago-test.py#L131). Otherwise, the fixture was being appended to the failure message.

Adding that comma to my code now produces this debug output:
```
Expecting success from: "spago test"
Command: 
['spago', 'test']
Failure message: 
Spago should test successfully
Expected output fixture: 
test-output.txt
```

**Comment by f-f** [comment]
@JordanMartinez thanks for debugging that! It looks like there's a logic bug in the `call` function: the fixture is only checked if the program doesn't exit successfully 😅

**Comment by JordanMartinez** [comment]
Changing `call()`'s usage of `check_output` to this:
```python
        result = subprocess.check_output(command, stderr=subprocess.STDOUT).decode('utf-8')
        print("No exception called...")
        return result
```
produces this (newlines added to break up tests better:
```
Expecting success from: "spago init"
No exception called...

Expecting failure from: "spago init"

Expecting success from: "rm spago.dhall packages.dhall"
No exception called...

Expecting success from: "spago init"
No exception called...

Expecting success from: "rm src/Main.purs"
No exception called...

Expecting success from: "rmdir src"
No exception called...

Expecting success from: "spago init -f"
No exception called...

Expecting success from: "spago init -f"
No exception called...
Successfully verified fixture for "spago-psc-success.dhall"

Expecting success from: "spago init -f"
No exception called...
Successfully verified fixture for "spago-psc-failure.dhall"

Going to run this for 0.5s: "spago install -j 3"
Expecting success from: "spago install"
No exception called...

Expecting success from: "spago install simple-json foreign"
No exception called...
Successfully verified fixture for "spago-install-success.dhall"

Expecting failure from: "spago install foobar"
No exception called...
Successfully verified fixture for "spago-install-failure.dhall"

Expecting success from: "spago build"
No exception called...

Expecting success from: "spago build -- -o myOutput"
No exception called...

Expecting success from: "spago build --path another_source_path/*.purs"
No exception called...

Expecting success from: "spago test"
No exception called...

Expecting success from: "spago bundle --to bundle.js"
No exception called...
Successfully verified fixture for "bundle.js"

Expecting success from: "spago make-module --to module.js"
No exception called...
Successfully verified fixture for "module.js"

Expecting success from: "rm -rf .spago src test packages.dhall spago.dhall bundle.js module.js output myOutput another_source_path psc-package.json spago-psc-success.dhall spago-psc-failure.dhall spago-install-success.dhall spago-install-failure.dhall"
No exception called...
```

**Comment by JordanMartinez** [comment]
I'm going to change the issue to `test fixtures aren't checked`

**Comment by JordanMartinez** [comment]
On second thought, it appears the other fixtures are using a different mechanism. So, it really is just the test command's fixtures.

**Comment by f-f** [comment]
This was fixed in #141
